packages/connectivity/android/src/main/java/io/flutter/plugins/connectivity/ConnectivityPlugin.java

```diff
@@ -106,6 +112,56 @@ private String getNetworkTypeLegacy(ConnectivityManager manager) {
     }
   }
 
+  private static String getNetworkSubType(ConnectivityManager manager) {
+    NetworkInfo info = manager.getActiveNetworkInfo();
+
+    if (info == null || !info.isConnected())
+      return "none";
+
+    switch (info.getSubtype()) {
+      case TelephonyManager.NETWORK_TYPE_1xRTT:
```

Comment:
Is there a source for this info? How about a link to it?

Reply:
i dont get you. Care to explain a bit more?

Reply:
How someone is supposed to know the mapping you did here? Do you have an http link that could be used as documentation for this?

Reply:
I suggest to fully expose each type as a enum to dart. So we don't lose any information here. Same for iOS.

Reply:
I pushed some updates. I added reference links for the mobile broadband speeds.